proposals/table-component.md

```diff
@@ -0,0 +1,221 @@
+# Table Component
+
+Component to render tabular data.
+
+## Features
+
+* Sorting
+
+## Currently Unsupported
+
+* Pagination
+* Filtering
+* Fixed row
+* Fixed column
+* Striped rows
+* Hoverable row styling
+* Multi-column sort
+* Row headers
+* Expandable row
```

Comment:
Are we planning to use `<table>`, `<tr>`, `<td>`... elements to implement our table? Expandable row is pretty hard to implement with those (I had an experience ending up with two different components - `table` and `expandable-table` and we always regretted to not build it as a single component from the start). If we want to be more future-proof, we could experiment with grid as a replacement. It would also help with responsiveness.

Reply:
good ideas. thanks for this

Reply:
One concern is regarding the behavior of `colspan` and `rowspan`. Can this be done easily with layout grid? I've had trouble implementing in the past using `display:table` and don't have too much experience with grid.

Also will `layout:fixed` and `layout:auto` be affected by using `grid` layout? I need the default behavior of having cells auto-size based on cell content.

One implementation of an expandable table uses native `<table>`, `<td>`, `<tr>` elements. This is in the `antd` table: https://ant.design/components/table/ which uses [rc-table](https://github.com/react-component/table) under the hood.

Reply:
> One concern is regarding the behavior of colspan and rowspan. Can this be done easily with layout grid?

Yes! CSS Grid has `span` keyword that you can use in `grid-row` and `grid-column` to span over rows/columns: https://css-tricks.com/snippets/css/complete-guide-grid/#article-header-id-27

> Also will layout:fixed and layout:auto be affected by using grid layout? I need the default behavior of having cells auto-size based on cell content.

This is probably where grid shines the most. You can combine absolute values, percentage and [`fr` units](https://css-tricks.com/snippets/css/complete-guide-grid/#fr-unit) so any type of layout should be possible.